vistautils/io_utils.py

```diff
@@ -148,6 +171,15 @@ def is_empty(self) -> bool:
         return os.path.getsize(self._path) == 0
 
 
+@attrs(slots=True, frozen=True, auto_attribs=True)
+class _FromByteSourceCharSource(CharSource):
```

Comment:
This was confusing at first until I realized it's a `CharSource` from a `ByteSource`. How about
```suggestion
class _CharSourceFromByteSource(CharSource):
```